src/commands/execute.ts

```diff
@@ -64,10 +64,13 @@ export default class Execute extends Command {
   }): Promise<void> {
     const { providerName, profileId, language } = args;
 
-    // TODO: resuse check from New command
     const providerJson = await resolveProviderJson(providerName, { userError });
 
-    await resolveProfileSource(profileId, { userError });
+    const profile = await resolveProfileSource(profileId, { userError });
+
+    const parsedProfileId = `${
+      profile.scope !== undefined ? `${profile.scope}/` : ''
+    }.${profile.name}`;
```

Comment:
If I understood correctly, you're pre-pending with `scope/` so the dot shouldn't be there?

```suggestion
    }${profile.name}`;
```

Reply:
Also, since it's also used in `map` command, I'd reuse it as a function